.azure/pipelines/ci.yml

```diff
@@ -16,48 +40,250 @@ jobs:
+
+# Test jobs
 - template: jobs/default-build.yml
   parameters:
-    jobName: Windows_Build
-    jobDisplayName: "Build and test: Windows"
+    condition: ne(variables['SkipTests'], 'true')
+    jobName: Windows_Test
+    jobDisplayName: "Test: Windows Server 2016 x64"
     agentOs: Windows
     buildScript: ./eng/scripts/cibuild.cmd
+    buildArgs: -test "/p:SkipIISBackwardsCompatibilityTests=true /p:SkipIISTests=true /p:SkipIISExpressTests=true /p:SkipIISForwardsCompatibilityTests=true"
     beforeBuild:
     - powershell: "& ./src/Servers/IIS/tools/UpdateIISExpressCertificate.ps1; & ./src/Servers/IIS/tools/update_schema.ps1"
       displayName: Setup IISExpress test certificates and schema
-    buildArgs: "/p:SkipIISBackwardsCompatibilityTests=true /p:SkipIISTests=true /p:SkipIISExpressTests=true /p:SkipIISForwardsCompatibilityTests=true"
 - template: jobs/default-build.yml
   parameters:
-    jobName: MacOs_Build
-    jobDisplayName: "Build and test: macOS"
+    condition: ne(variables['SkipTests'], 'true')
+    jobName: MacOs_Test
+    jobDisplayName: "Test: macOS 10.13"
     agentOs: macOs
+    poolVmImage: macOS-10.13
     buildScript: ./eng/scripts/cibuild.sh
+    buildArgs: --test
     beforeBuild:
     - bash: "./eng/scripts/install-nginx-mac.sh"
       displayName: Installing Nginx
 - template: jobs/default-build.yml
   parameters:
-    jobName: Linux_Build
-    jobDisplayName: "Build and test: Linux"
+    condition: ne(variables['SkipTests'], 'true')
+    jobName: Linux_Test
+    jobDisplayName: "Test: Ubuntu 16.04 x64"
     agentOs: Linux
+    poolVmImage: ubuntu-16.04
     buildScript: ./eng/scripts/cibuild.sh
+    buildArgs: --test
     beforeBuild:
     - bash: "./eng/scripts/install-nginx-linux.sh"
       displayName: Installing Nginx
 - template: jobs/iisintegration-job.yml
   parameters:
+    condition: ne(variables['SkipTests'], 'true')
     TestGroupName: IIS
     skipArgs: " /p:SkipIISBackwardsCompatibilityTests=true /p:SkipIISTests=false /p:SkipIISExpressTests=true /p:SkipIISForwardsCompatibilityTests=true"
 - template: jobs/iisintegration-job.yml
   parameters:
+    condition: ne(variables['SkipTests'], 'true')
     TestGroupName: IISExpress
     skipArgs: "/p:SkipIISBackwardsCompatibilityTests=true /p:SkipIISTests=true /p:SkipIISExpressTests=false /p:SkipIISForwardsCompatibilityTests=true"
 - template: jobs/iisintegration-job.yml
   parameters:
+    condition: ne(variables['SkipTests'], 'true')
     TestGroupName: IISForwardCompat
     skipArgs: "/p:SkipIISBackwardsCompatibilityTests=true /p:SkipIISTests=true /p:SkipIISExpressTests=true /p:SkipIISForwardsCompatibilityTests=false"
 - template: jobs/iisintegration-job.yml
   parameters:
+    condition: ne(variables['SkipTests'], 'true')
     TestGroupName: IISBackCompat
     skipArgs: "/p:SkipIISBackwardsCompatibilityTests=false /p:SkipIISTests=true /p:SkipIISExpressTests=true /p:SkipIISForwardsCompatibilityTests=true"
-- template: jobs/site-extensions-job.yml
```

Comment:
FYI @pakrym  - wasn't sure of the status of this job since there is still a separate pipeline for building the site-extension. This job is currently failing in PRs, so I removed it until we're ready to make site-extension build in the normal CI build.

Reply:
It's not failing on CI currently, how does it fail after this change?

Reply:
https://github.com/aspnet/AspNetCore/pull/7138

Reply:
Thanks, i'll merge that into this PR after you complete it.